### ISSUE TITLE
The market screen is not updated after an order is cancelled

### ISSUE THREAD

**Issue by neura-sx** [cryptonomex/graphene-ui#237]
The market screen does not refresh itself after the user cancels an order - the cancellation process goes as intended but after the dialog box is dismissed the screen does not update itself accordingly.

**Comment by svk31** [comment]
@bytemaster This is because I removed the optimistic update of the marketstore on cancelations, I can put it back but there's an issue with the TransactionConfirm modal making it so I can't get feedback on whether the transaction was successful or not. 

However, this manual optimistic update is only necessary because we do not get notifications from the market subscription that an order has been cancelled. Would it be possible to add that to the witness_node?

**Comment by svk31** [comment]
We can't put this back in as is because of the interaction with the TransactionConfirm modal, because of the issue I mentioned above and also because we can't tell if the user cancelled at the confirmation stage. The whole chaining of promises and return values inside the WalletDb is a bit hard to follow tbh. RIght now it all resolves on unlocking the wallet, then no more feedback. If the wallet is already unlocked, it resolves instantly

@jcalfee any suggestions on how to get  feedback from the TransactionConfirm process?

**Comment by valzav** [comment]
@svk31 you need to listen to TransactionConfirmStore events in order to know whether user canceled or not. In Transfer.jsx there is example - look at lines 69 and 90. The confirmation was canceled if the state is like:
{broadcasting: false,
broadcast: false,
included: false,
closed: true,
transaction: null,
broadcasted_transaction: null}
